nova/scheduler/filters/bigvm_filter.py

```diff
@@ -152,49 +136,36 @@ def _check_flavor_extra_specs(self, host_state, flavor,
                        'flavor_name': flavor.name})
             return False
 
-        host_fractions = set(
-            x.strip() for x in extra_specs[self._EXTRA_SPECS_KEY].split(','))
-        known_host_fractions = set(supported_fractions) & host_fractions
-        if not known_host_fractions:
-            LOG.warning('Flavor attribute %(specs_key)s does not have a '
-                        'supported value (%(specs_values)s). Cannot schedule '
-                        'on %(host_state)s.',
+        host_fractions = set()
+        for x in extra_specs[self._EXTRA_SPECS_KEY].split(','):
+            try:
+                ratio = float(x)
+                if 0.0 <= ratio <= 1.0:
+                    host_fractions.add(ratio * hypervisor_ram_mb)
+            except ValueError:
+                pass
```

Comment:
I'd write this as
```python
try:
  ratio = float(x)
except ValueError:
  continue
if 0 <= ratio <= 1:
  host_fractions.add(ratio * hypervisor_ram_mb)
```
to keep the section between try-except as short as necessary.

Reply:
The working replacement would be:
```python
ratio = None
try:
  ratio = float(x)
except ValueError:
  continue
if 0 <= ratio <= 1:
  host_fractions.add(ratio * hypervisor_ram_mb)
```
Is that also better? I disliked the initial variable declaration, but I don't care either way.

Reply:
Why is the initial declaration necessary?

Reply:
Ah I overlooked the `pass` needs to be replaced by `continue` that's why I got `UnboundLocalError`... My bad, will be fixed. I initially wrote it the way you suggested, but thought that Python didn't expose vars outside of `try`s after all — wrongly so.